src/sentry/spans/consumers/process/flusher.py

```diff
@@ -118,41 +119,50 @@ def produce(payload: KafkaPayload) -> None:
 
             while not stopped.value:
-                now = int(time.time()) + current_drift.value
-                flushed_segments = buffer.flush_segments(max_segments=max_flush_segments, now=now)
+                with metrics.timer("spans.buffer.flusher.loop_body"):
+                    now = int(time.time()) + current_drift.value
+                    flushed_segments = buffer.flush_segments(
+                        max_segments=max_flush_segments, now=now
+                    )
 
-                should_backpressure.value = len(flushed_segments) >= max_flush_segments * len(
-                    buffer.assigned_shards
-                )
+                    if len(flushed_segments) >= max_flush_segments * len(buffer.assigned_shards):
+                        if backpressure_since.value == 0:
+                            backpressure_since.value = int(time.time())
+                    else:
+                        backpressure_since.value = 0
 
-                if not flushed_segments:
-                    time.sleep(1)
-                    continue
-
-                for _, flushed_segment in flushed_segments.items():
-                    if not flushed_segment.spans:
-                        # This is a bug, most likely the input topic is not
-                        # partitioned by trace_id so multiple consumers are writing
-                        # over each other. The consequence is duplicated segments,
-                        # worst-case.
-                        metrics.incr("sentry.spans.buffer.empty_segments")
+                    if not flushed_segments:
+                        time.sleep(1)
                         continue
 
-                    spans = [span.payload for span in flushed_segment.spans]
+                    with metrics.timer("spans.buffer.flusher.produce"):
+                        for _, flushed_segment in flushed_segments.items():
+                            if not flushed_segment.spans:
+                                # This is a bug, most likely the input topic is not
+                                # partitioned by trace_id so multiple consumers are writing
+                                # over each other. The consequence is duplicated segments,
+                                # worst-case.
+                                metrics.incr("sentry.spans.buffer.empty_segments")
+                                continue
 
-                    kafka_payload = KafkaPayload(
-                        None, rapidjson.dumps({"spans": spans}).encode("utf8"), []
-                    )
+                            spans = [span.payload for span in flushed_segment.spans]
 
-                    metrics.timing("spans.buffer.segment_size_bytes", len(kafka_payload.value))
-                    produce(kafka_payload)
+                            kafka_payload = KafkaPayload(
+                                None, rapidjson.dumps({"spans": spans}).encode("utf8"), []
```

Comment:
While we're at it - most other code uses `orjson`. Can you double-check and update this, if it's valid?

Reply:
done